src/plugins/stake-plugins/thorchainSavers/tcSavers.ts

```diff
@@ -0,0 +1,168 @@
+  const fetchResponse = await fetchInfo(`v1/apyValues`)
+    .then(async res => {
+      if (!res.ok) {
+        throw new Error(`Fetch APY invalid response: ${await res.text()}`)
+      }
+      return res
+    })
+    .catch(err => {
+      console.warn(`Fetch APY failed: ${err.message}`)
+    })
+  if (fetchResponse != null) {
+    try {
+      const fetchResponseJson = await fetchResponse.json()
+      const infoServerResponse = asInfoServerResponse(fetchResponseJson)
+      updatePolicyApys(infoServerResponse)
+    } catch (err: any) {
+      console.warn(`Parsing Fetch APY failed: ${err.message}`)
+    }
+  }
+
+  const instance: StakePlugin = {
+    policies,
+    async fetchChangeQuote(request: ChangeQuoteRequest): Promise<ChangeQuote> {
+      const { action, stakePolicyId } = request
+      const policy = getPolicyFromId(stakePolicyId)
+
+      return changeQuoteFuncs[action](request, policy)
+    },
+    async fetchStakePosition(request: StakePositionRequest): Promise<StakePosition> {
+      const { stakePolicyId } = request
+      const policy = getPolicyFromId(stakePolicyId)
+      const { pluginId, currencyCode } = policy.stakeAssets[0]
+      return {
+        allocations: [
+          {
+            pluginId,
+            currencyCode,
+            allocationType: 'staked',
+            nativeAmount: '100000000'
+          },
+          {
+            pluginId,
+            currencyCode,
+            allocationType: 'earned',
+            nativeAmount: '100000'
+          }
+        ],
+        canStake: true,
+        canUnstake: true,
+        canClaim: true
+      }
+    }
+  }
+  return instance
+}
+
+const updatePolicyApys = (infoServerResponse: InfoServerResponse) => {
+  policies.forEach(policy => {
+    const apy = infoServerResponse.policies[policy.stakePolicyId]
+    if (apy != null) {
+      policy.apy = apy
+    }
+  })
+}
+
+const getPolicyFromId = (policyId: string): StakePolicy => {
+  const policy = policies.find(policy => policy.stakePolicyId === policyId)
+  if (policy == null) throw new Error(`Cannot find policy ${policyId}`)
+  return policy
+}
+
+const stakeRequest = async (request: ChangeQuoteRequest, policy: StakePolicy): Promise<ChangeQuote> => {
+  const { pluginId, currencyCode } = policy.stakeAssets[0]
+
+  return {
+    allocations: [
+      {
+        allocationType: 'stake',
+        pluginId,
+        currencyCode,
+        nativeAmount: '0'
+      }
+    ],
+    approve: async () => {}
+  }
+}
+
+const unstakeRequest = async (request: ChangeQuoteRequest, policy: StakePolicy): Promise<ChangeQuote> => {
+  const { pluginId, currencyCode } = policy.stakeAssets[0]
+
+  return {
+    allocations: [
+      {
+        allocationType: 'unstake',
+        pluginId,
+        currencyCode,
+        nativeAmount: '0'
+      }
+    ],
+    approve: async () => {}
+  }
+}
+
+const changeQuoteFuncs = {
+  stake: stakeRequest,
+  unstake: unstakeRequest,
+  claim: unstakeRequest
+}
```

Comment:
Hmm. I would like to move these private methods into the scope of the instance, but I understand this means testing these functions will requires async calls in order to instantiate the plugin object. Maybe we could keep the constrain that JavaScript class constructors have: sync only. This of course means reverting the method call to get stake policies, but I see `loadPolicies` method could take-place-of initializing at instantiation time.

I'm open to discussing this more. My motivation is to keep code more contiguous and maintainable by using things like closures. I think it's easier to reason about having class methods close over the instance state, even when writing unit tests (instantiating before testing methods is more true to real-world usage, so more reliable testing, IMO).

Reply:
My desire to keep these funcs outside the instance is to make them independent of instance variables so that I could move these function to another file in the future. Instance functions cause bloat as you can't move functions to another file so unless they are heavily dependent on instance params, I like to keep them out and make them pure functions which as you mentioned are easier to unit test.